gcs/src/main/scala/blobstore/gcs/GcsStore.scala

```diff
@@ -108,7 +105,7 @@ class GcsStore[F[_]: ConcurrentEffect: ContextShift](
     Stream.eval(Fs2OutputStream[F](chunkSize, maxChunksInFlight)).flatMap { os =>
       os.stream.concurrently(
         Stream.eval(
-          ConcurrentEffect[F].guarantee(blocker.delay(blob.downloadTo(os)))(Sync[F].delay(os.close()))
+          ConcurrentEffect[F].guarantee(blocker.delay(blob.downloadTo(os)))(ConcurrentEffect[F].delay(os.close()))
```

Comment:
I think it's better to restrict ourselves to the least powerful abstraction for suspension here. If we summon the `Sync` instance instead, it's easier to see what we need `ConcurrentEffect` for. E.g. replace the context bound with `Sync` and see what breaks.